### PR TITLE
HTS-1022 Added logic to UCThresholdActor to handle getting threshold …

### PR DESCRIPTION
**Pull request by Jackie1988**
…from DES

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -219,11 +219,12 @@ nsi.get-account {
 
 # let min-backoff be `m` and max-backoff be `M`. Then `2m-M` cannot equal zero
 uc-threshold {
-  # this is a temporary flag to disable the DES theshold behaviour until we are ready in productino
-  enabled     = true
+  # this is a temporary flag to disable the DES theshold behaviour until we are ready in production
+  enabled     = false
```
Comment (AndyHWChung) [inline conf/application.conf]: @MarcusQA is it more convenient to have the new threshold retrieval behaviour enabled or disabled by default?

File: app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala [hunk 2]
```diff
@@ -54,44 +56,87 @@ class UCThresholdManager(thresholdConnectorProxyActor: ActorRef,
...
-            }
-        }, {
-          value ⇒
-            logger.info(s"Successfully obtained the UC threshold value $value from DES")
-            context become ready(value)
-        }
-      )
+      getValueFromDES(None) pipeTo self
+
+    case GetThresholdValue ⇒
```
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala]: `GetThresholdValue` is the message which the `UCThresholdManager` listens out for and `GetDESThresholdValue` is the message which the `UCThresholdConnectorProxyActor` listens out for

[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala]: the `UCThresholdManager` also is re-using `GetDESThresholdValue` and listening out for those message to trigger certain behaviour

[the remark this answers is not in the document]
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala]: yeah, but it would be nice to make it clear in those names

**Pull request merged**